lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/TableInfo.java

```diff
@@ -137,6 +137,23 @@ public String getTranslatedTableAlias() {
 			return RowMetadataColumnTranslationReference.lookupColumnReference(rhs);
 		}
 	}
+	
+	/**
+	 * Does the passed TableNameCorrelation match this TableInfo?
+	 * 
+	 * @param tableNameCorrelation
+	 * @return
+	 */
+	public boolean isMatch(TableNameCorrelation tableNameCorrelation) {
+		String tableNameSql = tableNameCorrelation.getTableName().toSql();
```

Comment:
We might need to match on the alias too since the same table might be referenced more than once (in which case it would match always on the first TableInfo)